### PR TITLE
build: fix CI and simplify dev setup

### PR DESCRIPTION
**Pull request by Agrendalath**
This:
1. Drops support for Python 3.5 and Ubuntu 18.04. From now on, the CI will run on Python 3.8 and Ubuntu 20.04 because this is what `edx-platform` is using.
2. Fixes the CI by adding delays to flaky tests.
3. Simplifies the development setup. Previously, developers had to copy-paste instructions from different sources. Now, everything is handled by `make requirements`.
4. Removes `ddt` and `mock` from `setup.py`. These libraries are used only for tests, so we don't need them in the production environment.

## Author's notes
1. Unfortunately, it doesn't upgrade the Django version used for testing. Upgrading it above `1.11` causes inconsistent test failures, and will require more work.
2. This uses an ugly hack around `xblock-sdk` - it's documented in `requirements.txt`, but I wanted to highlight it here too.

## Other information
Private-ref: [BB-5562](https://tasks.opencraft.com/browse/BB-5562)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Agrendalath!

As a core committer in this repo, you can merge this once the pull request is approved per the [core committer reviewer requirements](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1529675973/Open+edX+Core+Committers#Reviews-and-Ownership) and according to the agreement with your edX Champion.

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Agrendalath!

As a core committer in this repo, you can merge this once the pull request is approved per the [core committer reviewer requirements](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1529675973/Open+edX+Core+Committers#Reviews-and-Ownership) and according to the agreement with your edX Champion.

File: tests/integration/test_render.py
Comment (Agrendalath) [inline tests/integration/test_render.py:37]: These styling changes in code are related only to upgrading pylint.

File: run_tests.py
Comment (Agrendalath) [inline run_tests.py:29]: The newer version of `xblock-sdk` was throwing these warnings after every single test, making them unreadable.

**Comment by Agrendalath** [comment]
@natabene, this only affects test pipelines and developer experience. Please let me know if you'd like to review it too, though.

**Review by abodacs — COMMENTED** [review]
Thanks for making the testing experience easier @Agrendalath

**Review by farhaanbukhsh — APPROVED** [review]
👍 

- ✅ I tested this by looking at the CI jobs and reading through the code
- ✅  I read through the code
- ❌  I checked for accessibility issues
- ❌  Includes documentation
- ❌  I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.

File: requirements.txt
Comment (Agrendalath) [reply requirements.txt:12]: @farhaanbukhsh, good catch! I knew that this line was suspiciously long…

Regarding `0.1.9` vs `v0.1.9` - this is not a typo - the usage of the `v` prefix is inconsistent [in these tags](https://github.com/openedx/xblock-sdk/releases).

**Comment by natabene** [comment]
@jmbowman Do you want to review before it is merged?

**Comment by jmbowman** [comment]
@natabene Tim's review is fine with me, feel free to merge.

**Comment by Agrendalath** [comment]
@timmc-edx,
> Could also switch over to pip-tools at some point, I think?

Yep. once we upgrade Django version here, it will unblock other dependencies, so `pip-tools` will be a nice option.

**Comment by openedx-webhooks** [comment]
@Agrendalath 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/xblock-drag-and-drop-v2&entry.867055334=https://github.com/openedx/xblock-drag-and-drop-v2/pull/293&entry.1484655318=https://github.com/Agrendalath&entry.752974735=2022-09-23+12:28&entry.1917517419=2022-09-28+19:02&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by openedx-webhooks** [comment]
@Agrendalath 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/xblock-drag-and-drop-v2&entry.867055334=https://github.com/openedx/xblock-drag-and-drop-v2/pull/293&entry.1484655318=https://github.com/Agrendalath&entry.752974735=2022-09-23+12:28&entry.1917517419=2022-09-28+19:02&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
